elastic-agent-client.proto

```diff
@@ -292,12 +292,22 @@ message CheckinAgentInfo {
     bool snapshot = 3;
 }
 
+message Features {
+    FQDNFeature fqdn = 1;
+}
+
+message FQDNFeature {
+    bool enabled = 1;
+}
+
 // A set of units and their expected states and configuration.
 message CheckinExpected {
     // Units is the expected units the component should be running.
     repeated UnitExpected units = 1;
     // Agent info is provided only on first CheckinExpected response to the component.
     CheckinAgentInfo agent_info = 2;
+
+    Features features = 3;
```

Comment:
Add comments explaining what this is, what the format is, and then document the FQDN feature itself and what clients should do with it please :)